### PR TITLE
Set JVM memory options for server applications

### PR DESCRIPTION
**Pull request by havocp**

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by henrikengstrom** [comment]
Nice one!

File: launch/src/main/scala/xsbt/boot/ServerApplication.scala [hunk 1]
```diff
@@ -211,6 +208,63 @@ object ServerLauncher {
...
+    pb.command("java", "-version")
+    pb.directory(currentDirectory)
+    val process = pb.start()
+    try {
+      process.getOutputStream.close()
+      process.getInputStream.close()
+      val stderr = new java.io.LineNumberReader(new java.io.InputStreamReader(process.getErrorStream))
+      // Looking for the first line which is `java version "1.7.0_60"` or similar
+      val lineOption = try Option(stderr.readLine()) finally stderr.close()
+      val re = """java version "[0-9]+\.([0-9]+)\..*".*""".r
```
Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala]: Sorry to do this, but did you check how much larger the launcher gets if you add in regex patterns like this?

**Comment by jsuereth** [comment]
Nice, this LGTM, caveat the one check on JAR size, and travis CI passing.

File: launch/src/main/scala/xsbt/boot/ServerApplication.scala (hunk 1 above)
Comment (havocp) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala]: no - I'm sure I could just use the plain java regex API, is that better?

**Comment by havocp** [comment]
Fixed up to use java Pattern and also to ignore (but print) errors from java -version.

**Comment by havocp** [comment]
Travis appears to be running out of ram. Don't know if any of those failures are "real"

**Comment by eed3si9n** [comment]
Restarted travis job.

**Pull request merged**